homeassistant/components/template/sensor.py

```diff
@@ -149,11 +230,25 @@ def __init__(
             icon_template=icon_template,
             entity_picture_template=entity_picture_template,
         )
-        self.entity_id = async_generate_entity_id(
-            ENTITY_ID_FORMAT, device_id, hass=hass
-        )
-        self._name = friendly_name
+        if object_id is not None:
+            self.entity_id = async_generate_entity_id(
+                ENTITY_ID_FORMAT, object_id, hass=hass
+            )
+
+        self._name: str | None = None
         self._friendly_name_template = friendly_name_template
+
+        if friendly_name_template:
+            friendly_name_template.hass = hass
+            try:
+                self._name = friendly_name_template.async_render(parse_result=False)
+            except template.TemplateError:
+                pass
+            finally:
+                # This is so TemplateEntity can group
```

Comment:
@bdraco templates are grouped during initialization by the `TemplateEntity`. But if we decide to try to render the name earlier, it is not grouped with the other templates because `hass` is not set on the others, so they are never equal, even if they are. 

I worked around it here by setting `hass` again to `None`. An alternative is that we always set `hass` inside `TemplateEntity.add_template_attribute`. What do you think is a better approach?

Reply:
I think its cleaner to do it in `TemplateEntity.add_template_attribute` since its only doing it just a bit sooner than https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/event.py#L783 will

Reply:
Done in #49094